### ISSUE TITLE
Error on retrieving record #-1:11273 (cluster: null)

### ISSUE THREAD

**Issue by gemmell** [orientechnologies/orientdb#3789]
---------- [OHazelcastPlugin]
2015-03-23 04:13:49:962 INFO  [Sydney] Saving distributed configuration file for database 'Ice' to: /opt/orientdb/databases/Ice/distributed-config.json [OHazelcastPlugin]
2015-03-23 04:13:50:126 INFO  Sent updated cluster configuration to the remote client 172.17.0.6:56241 [OClientConnectionManager]
2015-03-23 04:13:50:127 INFO  Sent updated cluster configuration to the remote client 172.17.0.6:56240 [OClientConnectionManager]
2015-03-23 04:13:50:447 INFO  [Sydney]<-[California] received updated status California.Ice=ONLINE [OHazelcastPlugin]Cannot route READ_RECORD operation for #-1:11273 to the distributed node
java.lang.IllegalArgumentException: Cannot read record #-1:11273 since the position is invalid in database 'Ice'
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:1414)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:694)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage$5.call(ODistributedStorage.java:543)
        at com.orientechnologies.orient.server.distributed.ODistributedAbstractPlugin.runInDistributedMode(ODistributedAbstractPlugin.java:75)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:540)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1572)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:80)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1437)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.readRecord(ONetworkProtocolBinary.java:1453)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:346)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Sent run-time exception to the client /172.17.0.6:56241: com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #-1:11273 (cluster: null)
com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #-1:11273 (cluster: null)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1605)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:80)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1437)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.readRecord(ONetworkProtocolBinary.java:1453)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:346)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Caused by: com.orientechnologies.orient.core.exception.OStorageException: Cannot route READ_RECORD operation for #-1:11273 to the distributed node
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.handleDistributedException(ODistributedStorage.java:1293)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:563)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1572)
        ... 6 more
Caused by: java.lang.IllegalArgumentException: Cannot read record #-1:11273 since the position is invalid in database 'Ice'
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:1414)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:694)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage$5.call(ODistributedStorage.java:543)
        at com.orientechnologies.orient.server.distributed.ODistributedAbstractPlugin.runInDistributedMode(ODistributedAbstractPlugin.java:75)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:540)
        ... 7 more
Cannot route READ_RECORD operation for #-1:11312 to the distributed node
java.lang.IllegalArgumentException: Cannot read record #-1:11312 since the position is invalid in database 'Ice'
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:1414)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:694)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage$5.call(ODistributedStorage.java:543)
        at com.orientechnologies.orient.server.distributed.ODistributedAbstractPlugin.runInDistributedMode(ODistributedAbstractPlugin.java:75)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:540)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1572)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:80)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1437)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.readRecord(ONetworkProtocolBinary.java:1453)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:346)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Sent run-time exception to the client /172.17.0.6:56241: com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #-1:11312 (cluster: null)
com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #-1:11312 (cluster: null)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1605)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:80)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1437)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.readRecord(ONetworkProtocolBinary.java:1453)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:346)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Caused by: com.orientechnologies.orient.core.exception.OStorageException: Cannot route READ_RECORD operation for #-1:11312 to the distributed node
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.handleDistributedException(ODistributedStorage.java:1293)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:563)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1572)
        ... 6 more
Caused by: java.lang.IllegalArgumentException: Cannot read record #-1:11312 since the position is invalid in database 'Ice'
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:1414)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:694)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage$5.call(ODistributedStorage.java:543)
        at com.orientechnologies.orient.server.distributed.ODistributedAbstractPlugin.runInDistributedMode(ODistributedAbstractPlugin.java:75)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:540)
        ... 7 more
Cannot route READ_RECORD operation for #-1:11273 to the distributed node
java.lang.IllegalArgumentException: Cannot read record #-1:11273 since the position is invalid in database 'Ice'
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:1414)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:694)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage$5.call(ODistributedStorage.java:543)
        at com.orientechnologies.orient.server.distributed.ODistributedAbstractPlugin.runInDistributedMode(ODistributedAbstractPlugin.java:75)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:540)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1572)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:80)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1437)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.readRecord(ONetworkProtocolBinary.java:1453)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:346)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Sent run-time exception to the client /172.17.0.6:56240: com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #-1:11273 (cluster: null)
com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #-1:11273 (cluster: null)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1605)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:80)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1437)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.readRecord(ONetworkProtocolBinary.java:1453)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:346)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Caused by: com.orientechnologies.orient.core.exception.OStorageException: Cannot route READ_RECORD operation for #-1:11273 to the distributed node
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.handleDistributedException(ODistributedStorage.java:1293)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:563)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1572)
        ... 6 more
Caused by: java.lang.IllegalArgumentException: Cannot read record #-1:11273 since the position is invalid in database 'Ice'
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:1414)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:694)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage$5.call(ODistributedStorage.java:543)
        at com.orientechnologies.orient.server.distributed.ODistributedAbstractPlugin.runInDistributedMode(ODistributedAbstractPlugin.java:75)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:540)
        ... 7 more
Cannot route READ_RECORD operation for #-1:11312 to the distributed node
java.lang.IllegalArgumentException: Cannot read record #-1:11312 since the position is invalid in database 'Ice'
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:1414)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:694)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage$5.call(ODistributedStorage.java:543)
        at com.orientechnologies.orient.server.distributed.ODistributedAbstractPlugin.runInDistributedMode(ODistributedAbstractPlugin.java:75)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:540)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1572)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:80)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1437)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.readRecord(ONetworkProtocolBinary.java:1453)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:346)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Sent run-time exception to the client /172.17.0.6:56240: com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #-1:11312 (cluster: null)
com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #-1:11312 (cluster: null)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1605)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:80)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1437)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.readRecord(ONetworkProtocolBinary.java:1453)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:346)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Caused by: com.orientechnologies.orient.core.exception.OStorageException: Cannot route READ_RECORD operation for #-1:11312 to the distributed node
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.handleDistributedException(ODistributedStorage.java:1293)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:563)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1572)
        ... 6 more
Caused by: java.lang.IllegalArgumentException: Cannot read record #-1:11312 since the position is invalid in database 'Ice'
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:1414)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:694)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage$5.call(ODistributedStorage.java:543)
        at com.orientechnologies.orient.server.distributed.ODistributedAbstractPlugin.runInDistributedMode(ODistributedAbstractPlugin.java:75)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:540)

**Comment by gemmell** [comment]
```
{"@type":"d","@version":0,"version":40,"autoDeploy":true,"hotAlignment":true,"executionMode":"asynchronous","readQuorum":1,"writeQuorum":2,"failureAvailableNodesLessQuorum":false,"readYourWrites":true,
```

**Comment by gemmell** [comment]
Built from 2.0.x branch of git.

**Comment by gemmell** [comment]
This is different to our others because we are getting this on a load from the database rather than a write. A single action by us causes:
```
---------- [OHazelcastPlugin]
2015-03-23 04:28:43:018 INFO  [Sydney] Saving distributed configuration file for database 'Ice' to: /opt/orientdb/databases/Ice/distributed-config.json [OHazelcastPlugin]
2015-03-23 04:28:43:030 INFO  [Sydney] found no previous messages in queue orientdb.node.Sydney.Ice.request [OHazelcastDistributedMessageService]
2015-03-23 04:28:44:071 INFO  {db=Ice} [Sydney] Installing distributed storage on database 'Ice' [ODistributedStorage]
2015-03-23 04:28:44:075 INFO  [Sydney] Publishing online status for database Sydney.Ice... [OHazelcastDistributedDatabase]
2015-03-23 04:28:44:077 INFO  [Sydney] received new status Sydney.Ice=ONLINE [OHazelcastPlugin]
2015-03-23 04:28:44:082 WARNING [Sydney] updated node status to 'ONLINE' [OHazelcastPlugin]
2015-03-23 04:28:44:083 INFO  Installing Script interpreter. WARN: authenticated clients can execute any kind of code into the server by using the following allowed languages: [sql] [OServerSideScriptInterpreter]
2015-03-23 04:28:44:084 INFO  OrientDB Server v2.0.6-SNAPSHOT is active. [OServer]
2015-03-23 04:28:49:684 INFO  [172.31.6.12]:5701 [orientdb] [3.3] Accepting socket connection from /192.168.1.54:56509 [SocketAcceptor]
2015-03-23 04:28:49:689 INFO  [172.31.6.12]:5701 [orientdb] [3.3] Established socket connection between /172.17.0.8:5701 and /192.168.1.54:56509 [TcpIpConnectionManager][ec2-user@sydneydb stuff]$ tail -f orientdb/log/orientdb.err.log
cat: invalid option -- 'f'
Try 'cat --help' for more information.
[ec2-user@sydneydb stuff]$ cat orientdb/l/ail^Cforientdb.err.log
[ec2-user@sydneydb stuff]$ tail -f orientdb/log/orientdb.err.log
2015-03-23 04:28:43:018 INFO  [Sydney] Saving distributed configuration file for database 'Ice' to: /opt/orientdb/databases/Ice/distributed-config.json [OHazelcastPlugin]
2015-03-23 04:28:43:030 INFO  [Sydney] found no previous messages in queue orientdb.node.Sydney.Ice.request [OHazelcastDistributedMessageService]
2015-03-23 04:28:44:071 INFO  {db=Ice} [Sydney] Installing distributed storage on database 'Ice' [ODistributedStorage]
2015-03-23 04:28:44:075 INFO  [Sydney] Publishing online status for database Sydney.Ice... [OHazelcastDistributedDatabase]
2015-03-23 04:28:44:077 INFO  [Sydney] received new status Sydney.Ice=ONLINE [OHazelcastPlugin]
2015-03-23 04:28:44:082 WARNING [Sydney] updated node status to 'ONLINE' [OHazelcastPlugin]
2015-03-23 04:28:44:083 INFO  Installing Script interpreter. WARN: authenticated clients can execute any kind of code into the server by using the following allowed languages: [sql] [OServerSideScriptInterpreter]
2015-03-23 04:28:44:084 INFO  OrientDB Server v2.0.6-SNAPSHOT is active. [OServer]
2015-03-23 04:28:49:684 INFO  [172.31.6.12]:5701 [orientdb] [3.3] Accepting socket connection from /192.168.1.54:56509 [SocketAcceptor]
2015-03-23 04:28:49:689 INFO  [172.31.6.12]:5701 [orientdb] [3.3] Established socket connection between /172.17.0.8:5701 and /192.168.1.54:56509 [TcpIpConnectionManager]Cannot route READ_RECORD operation for #-1:11273 to the distributed node
java.lang.IllegalArgumentException: Cannot read record #-1:11273 since the position is invalid in database 'Ice'
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:1414)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:694)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage$5.call(ODistributedStorage.java:543)
        at com.orientechnologies.orient.server.distributed.ODistributedAbstractPlugin.runInDistributedMode(ODistributedAbstractPlugin.java:75)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:540)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1572)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:80)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1437)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.readRecord(ONetworkProtocolBinary.java:1453)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:346)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Sent run-time exception to the client /172.17.0.9:60282: com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #-1:11273 (cluster: null)
com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #-1:11273 (cluster: null)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1605)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:80)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1437)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.readRecord(ONetworkProtocolBinary.java:1453)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:346)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Caused by: com.orientechnologies.orient.core.exception.OStorageException: Cannot route READ_RECORD operation for #-1:11273 to the distributed node
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.handleDistributedException(ODistributedStorage.java:1293)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:563)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1572)
        ... 6 more
Caused by: java.lang.IllegalArgumentException: Cannot read record #-1:11273 since the position is invalid in database 'Ice'
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:1414)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:694)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage$5.call(ODistributedStorage.java:543)
        at com.orientechnologies.orient.server.distributed.ODistributedAbstractPlugin.runInDistributedMode(ODistributedAbstractPlugin.java:75)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:540)
        ... 7 more
Cannot route READ_RECORD operation for #-1:11312 to the distributed node
java.lang.IllegalArgumentException: Cannot read record #-1:11312 since the position is invalid in database 'Ice'
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:1414)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:694)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage$5.call(ODistributedStorage.java:543)
        at com.orientechnologies.orient.server.distributed.ODistributedAbstractPlugin.runInDistributedMode(ODistributedAbstractPlugin.java:75)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:540)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1572)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:80)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1437)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.readRecord(ONetworkProtocolBinary.java:1453)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:346)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Sent run-time exception to the client /172.17.0.9:60282: com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #-1:11312 (cluster: null)
com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #-1:11312 (cluster: null)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1605)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:80)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1437)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.readRecord(ONetworkProtocolBinary.java:1453)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:346)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Caused by: com.orientechnologies.orient.core.exception.OStorageException: Cannot route READ_RECORD operation for #-1:11312 to the distributed node
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.handleDistributedException(ODistributedStorage.java:1293)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:563)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1572)
        ... 6 more
Caused by: java.lang.IllegalArgumentException: Cannot read record #-1:11312 since the position is invalid in database 'Ice'
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:1414)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.readRecord(OAbstractPaginatedStorage.java:694)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage$5.call(ODistributedStorage.java:543)
        at com.orientechnologies.orient.server.distributed.ODistributedAbstractPlugin.runInDistributedMode(ODistributedAbstractPlugin.java:75)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.readRecord(ODistributedStorage.java:540)
        ... 7 more
```

**Comment by gemmell** [comment]
Still happens with hotAlignment: false

**Comment by lvca** [comment]
By reading this log, seems the client send a LOAD command to the server to get the record #-1:11312, but this is a temporary record, not stored yet. Could you share the snippet of code that do that?

**Comment by lvca** [comment]
Fixed in 2.0.x branch.
